### PR TITLE
Migrate vexflow_test_helpers

### PR DESCRIPTION
**Pull request by ronyeh**
My migration of the Rests tests had some feature creep as I uncovered and fixed bugs and cleaned up related classes.

One of the files I cleaned up was `vexflow_test_helpers.` I broke it out into its own PR. The Rests tests is now rebased off of this work.

I would recommend merging this PR before continuing to migrate other test pages.

This PR uses eslint-disable liberally. Some of these anys will be removed in future PRs (after more test pages are migrated).

### CODE REVIEW

**Comment by ronyeh** [comment]
npm run test:reference shows no diffs relative to rev 73b01c61ea67c3a8539ec77eec424d2bc6ff5aba.

**Comment by ronyeh** [comment]
I also improved `runCanvasTest()` to support running tests with all three fonts. This was useful for me while debugging the Rests tests.

By default it is OFF, and `runCanvasTest()` will only test Bravura. Change `TEST_ALL_FONTS` to true and then reload flow.html to see all the fonts in the CanvasContext.

**Comment by ronyeh** [comment]
`npm run test:reference` shows no diffs relative to current master.

**Comment by ronyeh** [comment]
No visual diffs relative to master rev 3eea242fc47c7d540b226e969b6ad6a984ee5f7e

This one's ready to go!

File: tests/vexflow_test_helpers.ts [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+function useTempFontStack(fontName: string): void {
+  originalFontStack = VF.DEFAULT_FONT_STACK;
+  VF.DEFAULT_FONT_STACK = VexFlowTests.FONT_STACKS[fontName];
+}
+function restoreOriginalFontStack(): void {
+  VF.DEFAULT_FONT_STACK = originalFontStack;
+}
+
+// A micro util inspired by jQuery.
+if (!global.$) {
```
Comment (0xfe) [inline tests/vexflow_test_helpers.ts:36]: This is nice!

**Comment by 0xfe** [comment]
Awesome, thanks. Merging.

**Pull request merged**